File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # prompt user for input
    grade = get_string("Text: ")

    # storing value of string length
    length = len(grade)

    # calling letters function
    l = letters(length, grade)

    # calling words function
    w = words(length, grade)

    # calling sentences function
    s = sentences(length, grade)

    # calculator for the grade of text
    L = l / w * 100
    S = s / w * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    x = round(index)

    # printing the grade of the text
    if x >= 1 and x <= 16:
        print(f"Grade {x}")
    elif x < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")

# defining function for counting letters


def letters(a, b):
    c = 0
    for q in range(a):
        if (b[q] >= 'a' and b[q] <= 'z') or (b[q] >= 'A' and b[q] <= 'Z'):
            c += 1
    return c

# defining function for counting words


def words(a, b):
    c = 1
    for q in range(a):
        if (b[q] == ' '):
            c += 1
    return c

# defining function for counting sentences


def sentences(a, b):
    c = 0
    for q in range(a):
        if (b[q] == '.' or b[q] == '?' or b[q] == '!'):
            c += 1
    return c


main()

File: pset2/readability/readability.c
//libraries
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#include<math.h>

//defining the functioons to be used
int letters();
int words();
int sentences();

//main function
int main(void)
{
    //prompt user for input
    string grade = get_string("Text: ");

    //storing value of string length
    int length = strlen(grade);

    //calling letters function
    int l = letters(length, grade);

    //calling words function
    int w = words(length, grade);

    //calling sentences function
    int s = sentences(length, grade);

    //calculator for the grade of text
    float L = (float) l / w * 100;
    float S = (float) s / w * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int x = round(index);


    //printing the grade of the text
    if (x >= 1 && x <= 16)
    {
        printf("Grade %i\n", x);
    }
    else if (x < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }

}

//defining function for counting letters
int letters(int a, string b)
{
    int c = 0;
    for (int q = 0; q < a; q++)
    {

        if ((b[q] >= 'a' && b[q] <= 'z') || (b[q] >= 'A' && b[q] <= 'Z'))
        {
            c++;
        }
    }
    return c;
}

//defining function for counting words
int words(int a, string b)
{
    int c = 1;
    for (int q = 0; q < a; q++)
    {

        if (b[q] == ' ')
        {
            c++;
        }
    }
    return c;
}

//defining function for counting sentences
int sentences(int a, string b)
{
    int c = 0;
    for (int q = 0; q < a; q++)
    {

        if (b[q] == '.' || b[q] == '?' || b[q] == '!')
        {
            c++;
        }
    }
    return c;
}